File: 1.two-sum.py
# @leet start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash={}
        for i in range(0,len(nums)):
            hash[nums[i]]=i
        for i in range(0,len(nums)):
            temp = target -nums[i]
            if temp in hash and hash[temp]!=i:
                return [hash[temp],i]
        return [hash[temp],i]
        
# @leet end

File: 1.two-sum.cs
// @leet start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
		Dictionary<int, int> dict = new Dictionary<int, int>();
		for (int i = 0; i < nums.Length; i++) {
			int complement = target - nums[i];
			if (dict.ContainsKey(complement)) {
				return new int[] { dict[complement], i };
			}
			dict[nums[i]] = i;
		}
		throw new ArgumentException("No two sum solution");
		return null;
    }
}
// @leet end
